Ability to create AKS in existing subnet (#27)
epiphany-platform/m-azure-kubernetes-service: README.md

az account set --subscription="SUBSCRIPTION_ID"
az ad sp create-for-rbac --role="Contributor" --scopes="/subscriptions/SUBSCRIPTION_ID" --name="SOME_MEANINGFUL_NAME" #get appID, password, tenant, name and displayName
```
# Run module

> AKS and new subnet will be created in resource group and vnet from [AzBI Module](https://github.com/epiphany-platform/m-azure-basic-infrastructure) or you can create AKS in already existing subnet.

Reviewer: ```suggestion
AzKS and new subnet will be created in resource group and vnet from [AzBI Module](https://github.com/epiphany-platform/m-azure-basic-infrastructure) or you can create AzKS in already existing subnet.
```

* Initialize AzKS module in [AzBI Module](https://github.com/epiphany-platform/m-azure-basic-infrastructure)(without parameters it will extract some knowledge from status file):
  ```shell
  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/azks:latest init
  ```
  or initialize AzKS with some parameters (ie.: in already existing subnet by setting `M_SUBNET_NAME`): 
  ```shell
  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/azks:latest init M_RG_NAME="existing rg name" M_VNET_NAME="existing vnet name" M_SUBNET_NAME="existing subnet name"
  ```

>   This commad will create configuration file of AzKS module in /tmp/shared/azks/azks-config.yml. You can investigate what is stored in that file and change it at will.

Reviewer: ```suggestion
  Previous commad created configuration file of AzKS module in `/tmp/shared/azks/azks-config.yml`. You can investigate what is stored in that file and change it at will.
```

* Plan and apply AzKS module:
  ```shell
  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/azks:latest plan M_ARM_CLIENT_ID=appId M_ARM_CLIENT_SECRET=password M_ARM_SUBSCRIPTION_ID=subscriptionId M_ARM_TENANT_ID=tenantId
  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/azks:latest apply M_ARM_CLIENT_ID=appId M_ARM_CLIENT_SECRET=password M_ARM_SUBSCRIPTION_ID=subscriptionId M_ARM_TENANT_ID=tenantId
  ```

>   Running those commands should create AKS service. You should verify in Azure Portal.

Reviewer: ```suggestion
  Running those commands should create AzKS service. You should verify in Azure Portal.
```

# Run module

> AKS and new subnet will be created in resource group and vnet from [AzBI Module](https://github.com/epiphany-platform/m-azure-basic-infrastructure) or You can join AKS to already existing subnet.

Reviewer: ```suggestion
AKS and new subnet will be created in resource group and vnet from [AzBI Module](https://github.com/epiphany-platform/m-azure-basic-infrastructure) or you can create AKS in already existing subnet.
```

Author: Thanks Mateusz! Done! :)

> * Initialize AzKS module in AzBI:

Reviewer: ```suggestion
* Initialize AzKS module in AzBI (without parameters it will extract some knowledge from status file):
```

Author: Sure, done:)

  ```shell
  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/azks:latest init
  ```

>   or You can run AzKS module in already existing subnet: 

Reviewer: ```suggestion
  or you can run AzKS module with some parameters (ie.: in already existing subnet by setting `M_SUBNET_NAME`): 
```

Author: Sure, done:)

  ```shell
  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/azks:latest init M_RG_NAME="existing rg name" M_VNET_NAME="existing vnet name" M_SUBNET_NAME="existing subnet name"
  ```

>   This commad will create configuration file of AzKS module in /tmp/shared/azks/azks-config.yml. You can investigate what is stored in that file. 

Reviewer: ```suggestion
  This commad will create configuration file of AzKS module in /tmp/shared/azks/azks-config.yml. You can investigate what is stored in that file and change it at will. 
```

Author: Sure, done:)

* Plan and apply AzKS module:
  ```shell
  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/azks:latest plan M_ARM_CLIENT_ID=appId M_ARM_CLIENT_SECRET=password M_ARM_SUBSCRIPTION_ID=subscriptionId M_ARM_TENANT_ID=tenantId
  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/azks:latest apply M_ARM_CLIENT_ID=appId M_ARM_CLIENT_SECRET=password M_ARM_SUBSCRIPTION_ID=subscriptionId M_ARM_TENANT_ID=tenantId
  ```
  Running those commands should create AKS service. You should verify in Azure Portal.

> * Share kubeconfig with `epicli` tool:

Reviewer: ```suggestion

* Extract kubeconfig following way:
```

Author: Sure, done:)

  ```shell

>   docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/azks:latest kubeconfig M_ARM_CLIENT_ID=appId M_ARM_CLIENT_SECRET=password M_ARM_SUBSCRIPTION_ID=subscriptionId M_ARM_TENANT_ID=tenantId

Reviewer: ```suggestion
  docker run --rm -v /tmp/shared:/shared -t epiphanyplatform/azks:latest kubeconfig
```
I don't think azure credentials are required for this command but please double check.

Author: Sure, done:) Double check done, indeed the parameters are not necessary:) Removed:)

  ```
  This command will create file `/tmp/shared/kubeconfig`. You will need to move this file manually to `/tmp/shared/build/your-cluster-name/kubeconfig`. 

# Run example

> You will run example of AzBI and AzKS module

Reviewer: ```suggestion
You can run example of AzBI and AzKS module using files in `examples` directory. 
```

Author: Sure, done:)

make all
```

# Run example in existing subnet

You can run example of AzKS in existing subnet using files in `examples` directory.

```shell
cd examples/create_in_existing_subnet
ARM_CLIENT_ID="appId field" ARM_CLIENT_SECRET="password field" ARM_SUBSCRIPTION_ID="id field" ARM_TENANT_ID="tenant field" M_RG_NAME="existing rg name" M_SUBNET_NAME="existing subnet name" M_VNET_NAME="existing vnet name" EXISTING_SUBNET="true" make all
```

Or use config file with credentials:

```shell
cd examples/create_in_existing_subnet
cat >azure.mk <<'EOF'
ARM_CLIENT_ID ?= "appId field"
ARM_CLIENT_SECRET ?= "password field"
ARM_SUBSCRIPTION_ID ?= "id field"
ARM_TENANT_ID ?= "tenant field"
M_RG_NAME ?= "existing rg name"
M_SUBNET_NAME ?= "existing subnet name"
M_VNET_NAME ?= "existing vnet name"
EOF
make apply
```

> If You want to destroy the AKS, execute above instruction in the same way using `destroy` command instead of `all`.

Reviewer: ```suggestion
If You want to destroy the AzKS, execute above instruction in the same way using `destroy` command instead of `all`.
```

make all
```

# Run example in existing subnet

You will run example of AzKS module in existing subnet

```shell

> cd examples/basic_flow

Reviewer: I'd move this to `examples/create_in_existing_subnet

Author: Sure, done:)
